### DISCUSSION TITLE
Add index-url option to py-config as external source for .whl files

### DISCUSSION THREAD

**Discussion by msachsenhauser** [pyscript/pyscript#1201] [Proposals]
**Checklist**

- [x] added a descriptive title
- [x] searched for other issues and couldn't find a solution or duplication
- [x] Describe the behavior you observed, and the behavior you had expected.

**What happened?**

Inside a corporate network / behind a proxy you have to specify all urls to all relevant wheels, which makes the pyscript setup unfeasable.

**Proposal**

Would it be possible to add options to py-config to specify an alternative package-server? E.g. similar to pip.ini, by adding an "index-url" and maybe a "cert"/"trusted-host" option?

```html
<py-config>
packages = ["travertino"]
index_url = ["https://packageserver.corporate.network.com/artifactory/api/pypi/pypi-virtual/simple"]
trusted_host = ["packageserver.corporate.network.com"]
</py-config>
```

Thank you!

**Comment by JeffersGlass** [comment]
Interesting! Just so we understand better - it sounds like the issue is that corporate network and/or proxy are blocking some actions of micropip downloading packages from PyPI, but specifying their location by a specific wheel on PyPI does work (eg. a `files.pythonhosted.org` address)? Have I got that right?

Could you include either a copy/paste or image of the specific errors you see when the issue happens, so we can be sure we're specific in addressing it?

Thanks!

**Comment by msachsenhauser** [comment]
Basically, we are not allowed to fetch packages **directly** from sources outside of the corporate network. Instead we have to use an internal package proxy for severall reasons - therefore specifying `files.pythonhosted.org` doesn't work (but the link to the wheel on the internal server 'should' work, _unsure which platform the packages are requested for_).

**Expectation**:

When a .whl-file is requested and "index_url" was provided, this address should be used for the package-lookup and -download.

**Demo**:

For better understanding (_the console log was fetched from a webserver without CORS enabled, which blocked the requests_):

```text
...
[pyscript/pyodide] micropip install bokeh,numpy,panel==0.13.1
pyodide.asm.js:10 distutils already loaded from default channel
pyodide.asm.js:10 packaging already loaded from default channel
pyodide.asm.js:10 pyparsing already loaded from default channel
pyodide.asm.js:10 Loading bokeh, numpy, jinja2, markupsafe, pillow, python-dateutil, six, typing-extensions, pyyaml, bleach, webencodings, tqdm, setuptools, certifi
...
       GET https://cdn.jsdelivr.net/pyodide/v0.21.2/full/tqdm-4.64.0-py2.py3-none-any.whl net::ERR_FAILED 403
...
       TypeError: Failed to fetch
    at loadBinaryFile (pyodide.asm.js:10:114740)
    at downloadPackage (pyodide.asm.js:10:132813)
    at Object.loadPackage (pyodide.asm.js:10:136717)
...
```

instead:

```text
...
       GET https://packageserver.corporate.network.com/artifactory/<funky-path>/tqdm-4.64.0-py2.py3-none-any.whl
...
```
